src/js/yeldaWebchatInjector.js

```diff
@@ -8,6 +8,116 @@ import 'lg-share.js'
 
 import config from '../config'
 
+/**
+ * Check whether the string is url encoded
+ * @param {String} str
+ * @returns {Boolean}
+ */
+const isStrLikelyEncoded = (str) => {
+  // Check if the string contains any characters that would typically need encoding
+  const needsEncoding = /[?&=<>#%{}|\\\^~\[\]`;"\s]/.test(str)
+
+  // Check if the string contains any non-ASCII characters
+  const hasNonASCII = /[^\x00-\x7F]/.test(str);
+
+  // Check if the string contains any percent-encoded characters
+  const hasPercentEncoding = /%[0-9A-Fa-f]{2}/.test(str)
+
+  // Return true if any of the conditions are met
+  return needsEncoding || hasNonASCII || hasPercentEncoding
+}
+
+/**
+ * Parse the function string to get the function name & arguments
+ * @param {String} str
+ * @returns {Object|null} {functionName, args}
+ */
+const parseFunctionString = (str) => {
+  // Regular expression to match function name and its arguments
+  const regex = /((?:[a-zA-Z_$][\w$]*\.)*)([a-zA-Z_$][\w$]*)\(([^]*)\)/
+
+  // Executing the regex to get the match
+  const match = regex.exec(str)
+
+  if (!match) {
+    // If the string doesn't match the expected format
+    return null
+  }
+
```

Comment:
:pray: thanks for this big work

:gear: as asked in https://trello.com/c/R0vOO9Rn/2531-webchat-on-click-on-function-qr-the-function-code-is-called-by-the-chatplugin-js , can you please add some tests to test/test.js ? 

ex 

```
  describe('yeldaChat.parseFunctionString', () => {
   it('should return {functionName: "bla", args: ["test"] if passed "this.bla("test")', () => {
      expect(yeldaChat.parseFunctionString(this.bla("test").to.be.............
    })
    
    // .... and more complex values like 
    
     it('should return {functionName: "this.bla", args: [true, 3, "test"] if passed "this.bla(true, 3, "test", [1,2], {'x':'x'})', () => {
      expect(yeldaChat.parseFunctionString(this.bla(true, 3, "test", [1,2], {'x':'x'})).to.be.............
    })
   ```